engine/access/rpc/backend/connection_factory.go

```diff
@@ -136,12 +176,22 @@ func (cf *ConnectionFactoryImpl) GetExecutionAPIClient(address string) (executio
 	return executionAPIClient, nil
 }
 
-func (cf *ConnectionFactoryImpl) InvalidateExecutionAPIClient(address string) bool {
-	grpcAddress, err := getGRPCAddress(address, cf.ExecutionGRPCPort)
-	if err != nil {
-		return true
+func (cf *ConnectionFactoryImpl) InvalidateExecutionAPIClient(address string) {
+	cf.invalidateAPIClient(address, cf.ExecutionGRPCPort)
+}
+
+func (cf *ConnectionFactoryImpl) invalidateAPIClient(address string, port uint) {
+	grpcAddress, _ := getGRPCAddress(address, port)
+	if res, ok := cf.ConnectionsCache.Get(grpcAddress); ok {
+		store := res.(*ConnectionCacheStore)
+		store.mutex.Lock()
+		if store.ClientConn != nil {
+			conn := store.ClientConn
+			conn.Close()
```

Comment:
after looking at it some more, i don't particularly love the logic around how the invalidating of connections is handled. this is currently still a potential pain point where we're closing a connection without regard for whether there exists any threads that have acquired a client, and are still waiting to send a request out.

it would make more sense to add the logic to delay closing by the timeout time before actually attempting to close the connection